syntax = "proto3";

import "complex_types.proto";
service Nested {
  rpc NestedHello(SimpleValue) returns (SimpleValue);
  rpc DecimalOne(tsurugidb.udf.Decimal) returns (SimpleValue);
  rpc DecimalTwo(AddDecimal) returns (SimpleValue);
  rpc DateOne(tsurugidb.udf.Date) returns (tsurugidb.udf.Date);
  rpc DateTwo(AddDate) returns (tsurugidb.udf.Date);
  rpc LocalTimeOne(tsurugidb.udf.LocalTime) returns (tsurugidb.udf.LocalTime);
  rpc LocalTimeTwo(AddLocalTime) returns (tsurugidb.udf.LocalTime);
  rpc LocalDatetimeOne(tsurugidb.udf.LocalDatetime) returns (tsurugidb.udf.LocalDatetime);
  rpc LocalDatetimeTwo(AddLocalDatetime) returns (tsurugidb.udf.LocalDatetime);
  rpc OffsetDatetimeOne(tsurugidb.udf.OffsetDatetime) returns (tsurugidb.udf.OffsetDatetime);
  rpc BlobReferenceOne(tsurugidb.udf.BlobReference) returns (tsurugidb.udf.BlobReference);
  rpc ClobReferenceOne(tsurugidb.udf.ClobReference) returns (tsurugidb.udf.ClobReference);
}
message SimpleValue {
  string value = 1;
}
message AddDecimal {
  int32 i32 = 1;
  tsurugidb.udf.Decimal dec = 2;
}
message AddDate {
  string value = 1;
  tsurugidb.udf.Date dec = 2;
}

message AddLocalTime {
  string value = 1;
  tsurugidb.udf.LocalTime dec = 2;
  sint32 vv = 3;
}

message AddLocalDatetime {
  tsurugidb.udf.LocalDatetime dec = 2;
  sint32 vv = 3;
}
